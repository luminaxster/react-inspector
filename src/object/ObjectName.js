import PropTypes from 'prop-types';

import {propertyValueFormatter as defaultPropertyValueFormatter} from '../utils/propertyUtils';
import { useStyles } from '../styles';

/**
 * A view for object property names.
 *
 * If the property name is enumerable (in Object.keys(object)),
 * the property name will be rendered normally.
 *
 * If the property name is not enumerable (`Object.prototype.propertyIsEnumerable()`),
 * the property name will be dimmed to show the difference.
 */
const ObjectName = ({ name, dimmed = false, styles = {} , propertyValueFormatter = defaultPropertyValueFormatter}) => {
  const themeStyles = useStyles('ObjectName');
  const appliedStyles = {
    ...themeStyles.base,
    ...(dimmed ? themeStyles['dimmed'] : {}),
    ...styles,
  };

  return <span css={appliedStyles}>{propertyValueFormatter(name, 'any')}</span>;
};

ObjectName.propTypes = {
  /** Property name */
  name: PropTypes.string,
  /** Should property name be dimmed */
  dimmed: PropTypes.bool,
  /** Function to format the content. It receives the current object to be rendered and its type, it must return valid react children. */
  propertyValueFormatter: PropTypes.func,
};

export default ObjectName;
