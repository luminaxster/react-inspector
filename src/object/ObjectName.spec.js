import React from 'react';
import TestRenderer from 'react-test-renderer';
import ObjectName from './ObjectName';

describe('ObjectName', () => {
  it('should render', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" />);

    expect(tree).toMatchSnapshot();
  });

  it('should apply dimming if `dimmed` prop is true', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" dimmed={true} />);

    expect(tree).toMatchSnapshot();
  });

  it('should not apply dimming if `dimmed` prop is false', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" dimmed={false} />);

    expect(tree).toMatchSnapshot();
  });

  it('Accepts and applies additional `styles` prop', () => {
    // Test that a custom `styles` props is passed and applied to <span/>
    const styles = { color: 'hotpink' };
    const tree = TestRenderer.create(<ObjectName styles={styles} />);

    expect(tree).toMatchSnapshot();
  });
});
