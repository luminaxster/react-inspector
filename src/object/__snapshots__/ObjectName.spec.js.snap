// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectName Accepts and applies additional \`styles\` prop 1`] = `
.emotion-0 {
  color: hotpink;
}

<span
  className="emotion-0"
/>
`;

exports[`ObjectName should apply dimming if \`dimmed\` prop is true 1`] = `
.emotion-0 {
  color: rgb(136, 19, 145);
  opacity: 0.6;
}

<span
  className="emotion-0"
>
  testvalue
</span>
`;

exports[`ObjectName should not apply dimming if \`dimmed\` prop is false 1`] = `
.emotion-0 {
  color: rgb(136, 19, 145);
}

<span
  className="emotion-0"
>
  testvalue
</span>
`;

exports[`ObjectName should render 1`] = `
.emotion-0 {
  color: rgb(136, 19, 145);
}

<span
  className="emotion-0"
>
  testvalue
</span>
`;
